pkg/datastore/datastore.go

```diff
@@ -146,9 +163,9 @@ type RelationshipsFilter struct {
 	// If specified, relationships matching *any* selector will be returned.
 	OptionalSubjectsSelectors []SubjectsSelector
 
-	// OptionalCaveatName is the filter to use for caveated relationships, filtering by a specific caveat name.
-	// If nil, all caveated and non-caveated relationships are allowed
-	OptionalCaveatName string
+	// OptionalCaveatNameFilter is the filter to use for caveated relationships, filtering by a specific caveat name.
+	// If nil, relationships with and without caveats are returned.
+	OptionalCaveatNameFilter *CaveatNameFilter
```

Comment:
Why use a pointer? the zero value of `CaveatFilterOption` should be `CaveatFilterOptionNone`